tests/js/client/aql/aql-object-enumeration.js

```diff
@@ -0,0 +1,165 @@
+    }
+    const stmt = db._createStatement(query, {}, rules);
+    checkPlan(stmt.explain().plan, isOptimized);
+
+    return stmt.execute().toArray();
+  }
+
+  return {
+    setUpAll: function () {
+      db._createDatabase(database);
+      db._useDatabase(database);
+      db._create(docCollection);
+      db._createEdgeCollection(edgeColl);
+
+      db[docCollection].insert({"key1": "a"});
+      db[docCollection].insert({"key2": "b"});
+      db[docCollection].insert({"key3": "c"});
+      db[docCollection].insert({"key4": "e"});
+      db[docCollection].insert({"key5": "f"});
+      db[docCollection].insert({"key6": [1,2,3]});
+
+      db[edgeColl].insert({_from: docCollection + "/key1", _to: docCollection + "/key4", "field": {"a": 1}});
+      db[edgeColl].insert({_from: docCollection + "/key3", _to: docCollection + "/key4", "field": {"b": 2}});
+      db[edgeColl].insert({_from: docCollection + "/key5", _to: docCollection + "/key6", "field": {"c": 3}});
+    },
+
+    tearDownAll: function() {
+      db._useDatabase("_system");
+      db._dropDatabase(database);
+    },
+
+    testSimpleObjectIteration : function() {
+      const query = `
+        FOR doc IN ${docCollection}
+          FOR [key, value] IN ENTRIES(doc)
+          RETURN [key, value]
+      `;
+
+      const stmt = db._createStatement({query});
+      checkPlan(stmt.explain().plan);
+    },
+
+    testOutput1 : function() {
+
+
+      const query = `
+        FOR doc IN C
+          FOR [key, value] IN ENTRIES(doc)
+          RETURN [key, value]
+      `;
+
+      let actual = queryDocuments(docCollection);
+
+      // Check that system fields are present
+      let systemFields = actual.filter((e) => ["_key", "_rev", "_id"].includes(e[0]));
+      assertEqual(systemFields.length, 3 * 6); // 3 system fields * 6 documents in collection
+      actual = actual.filter((e) => !["_key", "_rev", "_id"].includes(e[0])).flat();
+      let expected = [ 
+        "key1", 
+        "a", 
+        "key2", 
+        "b", 
+        "key3", 
+        "c", 
+        "key4", 
+        "e", 
+        "key5", 
+        "f", 
+        "key6", 
+        [ 
+          1, 
+          2, 
+          3
+        ] 
+      ];
+      assertEqual(actual, expected);
+    },
+
+    testOutput2 : function() {
+      let actual = queryDocuments(edgeColl);
+
+      // Check that system fields are present
+      let systemFields = actual.filter((e) => ["_key", "_rev", "_id", "_from", "_to"].includes(e[0]));
+      assertEqual(systemFields.length, 5 * 3); // 5 system fields * 3 documents in collection
+      actual = actual.filter((e) => !["_key", "_rev", "_id", "_from", "_to"].includes(e[0])).flat();
+
+      let expected = [ 
+        "field", 
+        {"a": 1}, 
+        "field", 
+        {"b": 2}, 
+        "field", 
+        {"c": 3}
+      ];
+      assertEqual(actual, expected);
+    }
+  };
+}
+
+jsunity.run(enumerateObjectTestSuite);
```

Comment:
we should add tests, that skip some documents, i.e. with some `LIMIT 4, 3`. Furthermore we should also test that the full-count is computed correctly.